torch/nested/_internal/ops.py

```diff
@@ -547,12 +551,43 @@ def chunk_default(func, *args, **kwargs):
 
     inp = new_kwargs.pop("input")
 
-    new_kwargs["dim"] = _wrap_jagged_dim(inp.dim(), new_kwargs["dim"], "chunk")
+    new_kwargs["dim"] = _wrap_jagged_dim(
+        inp.dim(), new_kwargs["dim"], "chunk", allow_dim_zero=True
+    )
 
-    return [
-        NestedTensor(values=x, **extract_kwargs(inp))
-        for x in func(inp._values, **new_kwargs)
-    ]
+    if new_kwargs["dim"] == 0:
+        chunks = new_kwargs["chunks"]
+        dim0_size = inp._size[0]
+        chunk_size = math.ceil(dim0_size / chunks)
+
+        # get _offsets of the chunks
+        chunk_offsets = [
+            inp._offsets[i * chunk_size + 1 : (i + 1) * chunk_size + 1]
+            - inp._offsets[i * chunk_size]
+            for i in range(chunks)
+        ]
+        chunk_offsets = [F.pad(x, (1, 0), value=0) for x in chunk_offsets]
```

Comment:
Might be simpler if you converted to lengths, because then you can just apply chunk on the lengths as-is instead having to derive chunk_size yourself.
```
>>> offsets = torch.tensor([0, 4, 5, 9, 17])
>>> lengths = offsets.diff()
>>> lengths
tensor([4, 1, 4, 8])
>>> chunked_lengths = lengths.chunk(3)
>>> chunked_lengths
(tensor([4, 1]), tensor([4, 8]))
>>> torch.cat([torch.tensor([0]), torch.cumsum(chunked_lengths[0], dim=0)])
tensor([0, 4, 5])
```

Reply:
@soulitzer thanks for reviewing. Could you also explain why do you use `torch.cat` here? I used `F.pad` instead of `torch.cat` because it could avoid creating on demand tensor. (torch.cat would create the on demand tensor `torch.tensor([0])`

Reply:
You're right, we do create an extra tensor in this case. F.pad sounds good to me.